main.tf

```diff
@@ -11,111 +11,119 @@ module "database_label" {
 resource "snowflake_database" "this" {
   count = module.this.enabled ? 1 : 0
 
-  name    = local.name_from_descriptor
-  comment = var.comment
-
-  data_retention_time_in_days = var.data_retention_time_in_days
-  is_transient                = var.is_transient
-
-  from_database = var.from_database
-  from_replica  = var.from_replica
-  from_share    = var.from_share
+  name         = local.name_from_descriptor
+  is_transient = var.is_transient
+  comment      = var.comment
+
+  data_retention_time_in_days                   = var.data_retention_time_in_days
+  max_data_extension_time_in_days               = var.max_data_extension_time_in_days
+  external_volume                               = var.external_volume
+  catalog                                       = var.catalog
+  replace_invalid_characters                    = var.replace_invalid_characters
+  default_ddl_collation                         = var.default_ddl_collation
+  storage_serialization_policy                  = var.storage_serialization_policy
+  log_level                                     = var.log_level
+  trace_level                                   = var.trace_level
+  suspend_task_after_num_failures               = var.suspend_task_after_num_failures
+  task_auto_retry_attempts                      = var.task_auto_retry_attempts
+  user_task_managed_initial_warehouse_size      = var.user_task_managed_initial_warehouse_size
+  user_task_timeout_ms                          = var.user_task_timeout_ms
+  user_task_minimum_trigger_interval_in_seconds = var.user_task_minimum_trigger_interval_in_seconds
+  quoted_identifiers_ignore_case                = var.quoted_identifiers_ignore_case
+  enable_console_output                         = var.enable_console_output
 }
 
 module "snowflake_default_role" {
   for_each = local.default_roles
 
-  source  = "getindata/role/snowflake"
-  version = "1.4.0"
+  source  = "getindata/database-role/snowflake"
+  version = "1.1.1"
+  context = module.this.context
 
-  context         = module.this.context
-  enabled         = local.create_default_roles && each.value.enabled
-  descriptor_name = each.value.descriptor_name
+  database_name   = one(snowflake_database.this[*].name)
+  name            = each.key
+  enabled         = local.create_default_roles && lookup(each.value, "enabled", true)
+  descriptor_name = lookup(each.value, "descriptor_name", "snowflake-role")
+
+  granted_to_roles          = lookup(each.value, "granted_to_roles", [])
+  granted_to_database_roles = lookup(each.value, "granted_to_database_roles", [])
+  granted_database_roles    = lookup(each.value, "granted_database_roles", [])
+  database_grants           = lookup(each.value, "database_grants", {})
+  schema_grants             = lookup(each.value, "schema_grants", [])
 
-  name       = each.key
   attributes = [one(snowflake_database.this[*].name)]
```

Comment:
Pls - do not pass DB name to the role, since those will be DB roles, the DB name is implied :)